core/src/main/java/io/methvin/watcher/DirectoryWatcher.java

```diff
@@ -234,4 +243,15 @@ private void register(Path directory, boolean useFileTreeModifier) throws IOExce
     keyRoots.put(watchKey, directory);
   }
 
+  private void notifyCreateEvent(Path path, int count) throws IOException {
+    HashCode newHash = PathUtils.hash(path);
+    if (newHash != null && !pathHashes.containsKey(path)) {
+      logger.debug("{} [{}]", EventType.CREATE, path);
+      listener.onEvent(new DirectoryChangeEvent(EventType.CREATE, path, count));
+      pathHashes.put(path, newHash);
+    } else {
```

Comment:
I think this should be `else if (newHash == null)` since the log message is only relevant to that case.